plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Federico Olivieri (lvrfrc87@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: git_commit
+author:
+    - "Federico Olivieri (@Federico87)"
+version_added: "2.10"
+short_description: Perform git and and git commit operations.
+description:
+    - Manage git add and git commit on local git repository.
+options:
+    path:
+        description:
+            - Folder path where .git/ is located.
+        required: true
+        type: path
+    comment:
+        description:
+            - Git commit comment. Same as "git commit -m".
+        type: str
+        required: true
+    add:
+        description:
+            - list of files to be staged. Same as "git add ."
+              Asterisx values not accepted. i.e. "./*" or "*".
+        type: list
+        default: ["."]
+        elements: str
+requirements:
+    - git>=2.10.0 (the command line tool)
+'''
+
+EXAMPLES = '''
+- name: Add and commit 2 files.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+    add: ['test.txt', 'txt.test']
+
+- name: Add all files using default and commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+'''
+
+RETURN = '''
+output:
+    description: list of git cli command stdout
+    type: list
+    returned: always
+    sample: [
+        "[master 99830f4] Remove [ test.txt, tax.txt ]\n 4 files changed, 26 insertions(+)..."
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def git_commit(module):
+
+    commands = list()
+
+    add = module.params.get('add')
+    path = module.params.get('path')
+    comment = module.params.get('comment')
+
+    if add:
+        commands.append('git -C {path} add {add}'.format(
+            path=path,
+            add=' '.join(add),
+        ))
+
+    if comment:
+        commands.append('git -C {path} commit -m "{comment}"'.format(
+            path=path,
+            comment=comment,
+        ))
+
+    return commands
+
+
+def main():
+
+    argument_spec = dict(
+        path=dict(required=True, type="path"),
+        comment=dict(required=True),
+        add=dict(type="list", elements='str', default=["."]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    result_output = list()
+    result = dict(changed=False)
+
+    for cmd in git_commit(module):
+        _rc, output, error = module.run_command(cmd, check_rc=False)
+
+        if output:
+            if 'no changes added to commit' in output:
```

Comment:
What happens if output is localized?

For `git commit`, you should use the `--porcelain` option. No idea if something similar exists for `git add`.

Reply:
If I use `--porcelain` I am going to loose all the string outputs that I use for the `output` and `error` checking. rc in git are pretty useless as it pretty much always return `1` so I believe having strings to use as anchor for if statement is a good thing. If you have an idea how to implement a safe `output` `error` check using `--porcelain` please let me know. Here an example of what I mean:

```
(NaC) olivierif:NaC federicoolivieri$ git commit  -m "ciao"
On branch dev_infra
Your branch is ahead of 'origin/dev_infra' by 3 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
(NaC) olivierif:NaC federicoolivieri$ git commit  -m "ciao"  --porcelain
(NaC) olivierif:NaC federicoolivieri$
```

Reply:
Doing the output/error checks on string output is a terrible idea. This doesn't work at all.

Also please read `man git-commit` on what `--porcelain` does. In this case, empty output means "nothing to commit".

Reply:
Ok. What about for `git add` that does not support `--porcelain`. Is ok to use string output or do you have other ideas? I was thinking to use `rc` but I am not sure if distinguish between `nothing to commit, working tree clean` and fatal: pathspec 'file' did not match any files`

Reply:
You should run `git status --porcelain=v1`, like here: https://github.com/buildbot/buildbot/pull/5023/files#diff-380a2f2e80216ef42112b8b625db2516R753-R761

Then you know whether there's something to commit or not.

Parsing the normal text output of git is something you should NEVER rely on. There exist translated versions of CLI programs, and your code will do the wrong things when encountering such translated output.